hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -284,7 +288,20 @@ boolean isInternalDir() {
       return false;
     }
 
-    public T getTargetFileSystem() {
+    /**
+     * Gets lazily loaded instance of FileSystem
+     * @return An Initialized instance of T
+     * @throws IOException
+     */
+    public T getTargetFileSystem() throws IOException {
+      if (targetFileSystem != null)
+        return targetFileSystem;
+
+      if (targetDirLinkList.length == 1) {
+        synchronized (this) {
+          targetFileSystem = fileSystemInitFunc.apply(targetDirLinkList[0]);
```

Comment:
this will dual init as thread #2 will block. You need another check inside the sync block so the second thread won't repeat itself

Reply:
Added extra check